versioned_docs/version-V3/guides/governance/license-modifications.md

```diff
@@ -0,0 +1,155 @@
+
+These contract function calls should then be encoded into a governance proposal, and approved by Uniswap Governance.
+
+</details>
+
+## Proposals
+
+Proposals are submitted via `GovernorBravoDelegator` @ `0x408ED6354d4973f66138C91495F2f2FCbd8724C3`, a proxy contract currently pointing to the implementation at `0x53a328F4086d7C0F1Fa19e594c9b842125263026`. NPM packages for consuming the governance contract ABIs, and details on previous versions, are available [here](https://docs.uniswap.org/protocol/concepts/governance/overview).
+
+<details>
+    <summary> Governor Bravo #propose Parameters </summary>
+
+```solidity
+/**
+    * @notice Function used to propose a new proposal. Sender must have delegates above the proposal threshold
+    * @param targets Target addresses for proposal calls
+    * @param values Eth values for proposal calls
+    * @param signatures Function signatures for proposal calls
+    * @param calldatas Calldatas for proposal calls
+    * @param description String description of the proposal
+    * @return Proposal id of new proposal
+    */
+function propose(
+    address[] memory targets,
+    uint[] memory values,
+    string[] memory signatures,
+    bytes[] memory calldatas,
+    string memory description
+) public returns (uint)
+
+```
+
+</details>
+
+## Populating Proposal Calldata
+
+Below is an example of using a scripting environment to generate a proposal. This is for educational purposes only - that example assumes access to a private key with a sufficient amount of delegated UNI to submit a proposal, which is an insecure practice. There are several ways to generate a proposal transaction and submit it to Ethereum; this example should only be used for reference and not in production.
+
+<details>
+<summary> Populating `Propose` Calldata </summary>
+
+```typescript
+import { Contract, ethers } from 'ethers'
+import { namehash } from '@ethersproject/hash'
+import { keccak256 } from '@ethersproject/keccak256'
+import { Interface } from '@ethersproject/abi'
+// note: contract ABIs should be imported via etherscan
+import { GOVERNOR_BRAVO_ABI, ENS_REGISTRY_ABI, ENS_PUBLIC_RESOLVER_ABI } from './utils'
+
+const GOVERNOR_BRAVO_ADDRESS: string = '0x408ED6354d4973f66138C91495F2f2FCbd8724C3'
+const ENS_REGISTRY_ADDRESS: string = '0x00000000000C2E074eC69A0dFb2997BA6C7d2e1e'
+const PUBLIC_ENS_RESOLVER_ADDRESS: string = '0x4976fb03c32e5b8cfe2b6ccb31c09ba78ebaba41'
+
+const provider = new ethers.providers.JsonRpcProvider('YOUR_RPC_URL_HERE')
+const signer = provider.getSigner('YOUR_SIGNER_ADDRESS_HERE')
+
+const ensPublicResolverInterface = new Interface(ENS_PUBLIC_RESOLVER_ABI)
+const setTextCalldata = ensPublicResolverInterface.encodeFunctionData('setText', [
+  // node
+  namehash('v3-core-license-grants.uniswap.eth'),
+  // key
+  '[your-projects-additional-use-grant-title]',
+  // value
+  '[your-additional-use-grant-description]',
+])
+
+const ensRegistryInterface = new Interface(ENS_REGISTRY_ABI)
```

Comment:
i'd move this block above the other as it has to logically come first (which it does below in the call order)

also probably worth adding a comment noting that this part should only be done if the subdomain doesn't exist already